src/background.js

```diff
@@ -105,10 +117,116 @@ window.scrNoti.hasUnreadMessages = async () => {
 //==========================================
 // Notify the native script
 //==========================================
-window.scrNoti.notifyNativeScript = async (hasUnreadMessages) => {
+window.scrNoti.notifyNativeScript = async (message, event) => {
+  let payload = null;
+  const { scriptType } = await messenger.storage.local.get({
+    scriptType: "simple",
+  });
+
+  switch (scriptType) {
+    case "simple":
+      switch (event) {
+        case "new":
+          payload = true;
+          break;
+        case "read":
+        case "deleted":
+          payload = await window.scrNoti.hasUnreadMessages();
+          break;
+        case "start":
+          //==========================================
+          // For some reason, the folders may not be ready when
+          // Thunderbird starts (ex: "Error: Folder not found: /Inbox").
+          // So we retry for a couple of times before giving up.
+          //==========================================
+          payload = await window.scrNoti.tryNbrTimes(window.scrNoti.hasUnreadMessages, 10);
+          break;
+      };
+      break;
+    case "extended":
+      // List of all accounts
+      const accounts = await messenger.accounts.list(false);
+      const accountsList = {};
+      for (const account of accounts) {
+        const identitiesList = [];
+        for (const identity of account.identities) {
+          const mailIdentity = {
+            email: identity.email,
+            label: identity.label,
+            name: identity.name,
+            organization: identity.organization,
+          };
+          identitiesList.push(mailIdentity);
+        };
+
+        const mailAccount = {
+          identities: identitiesList,
+          name: account.name,
+          type: account.type,
+        };
+        accountsList[account.id] = mailAccount;
+      };
+
+      // List of all folders, which should be included
+      const foldersToInclude =
+        await window.scrNoti.getFoldersToCheckForUnread();
+      const foldersList = [];
+      for (const folder of foldersToInclude) {
+        const folderInfo = await messenger.folders.getFolderInfo(folder);
+        const folderData = {
+          accountId: folder.accountId,
+          favorite: folderInfo.favorite,
+          name: folder.name,
+          path: folder.path,
+          totalMessageCount: folderInfo.totalMessageCount,
+          type: folder.type,
+          unreadMessageCount: folderInfo.unreadMessageCount,
+          seenMessageCount: seenMessages[folder.accountId + folder.path].size,
+        };
+        foldersList.push(folderData);
+      };
+
+      // Message data
+      if (event == "start" || event == "quit") {
+        messageDetails = null;
```

Comment:
May it could be useful to send the result of `window.scrNoti.hasUnreadMessages` when Thunderbird starts?

Reply:
The extended payload has these fields for each watched folder:
- `totalMessageCount`
- `unreadMessageCount`
- `seenMessageCount`

so it would send an overall indicator, if `unreadMessageCount` is `>0` for any of the watched folders? Can certainly done. Do you think this would be a good idea? What would be the use-case? (I do not need it in my case, but that should not keep us from adding it.)

Reply:
The use case I see is if you start Thunderbird and there are unread messages, maybe you would want to do something? The "simple" mode does this, as it only check for unread messages, "seen" or not. 

But it was only an idea, don't implement it if you don't need it.

Reply:
Yes. At the moment you have to pull that information out of the `folders` dictionaries (and then get to you know, which folder has new messages).

I considered sending all new messages at startup to the script. I know people, who have 100s if not 1000s new messages in their inbox, which would slow down starting the add-on, so I decided against it.

To include the information as suggested would tell the script generally, that there are new messages available. It would not tell the script, in which folder. Should this be included? (I do not mind one way or the other.)

Reply:
Let's not and wait to see if someone ask for it one day. We can always improve everything later on, anyway.